src/main/java/net/neoforged/neoforge/client/extensions/common/IClientItemExtensions.java

```diff
@@ -142,17 +145,22 @@ default Model getGenericArmorModel(ItemStack itemStack, EquipmentClientInfo.Laye
     default void setupModelAnimations(LivingEntity livingEntity, ItemStack itemStack, EquipmentSlot equipmentSlot, Model model, float limbSwing, float limbSwingAmount, float partialTick, float ageInTicks, float netHeadYaw, float headPitch) {}
 
     /**
-     * Called when the client starts rendering the HUD, and is wearing this item in the helmet slot.
+     * Called to render an overlay on the first-person camera.
      * <p>
-     * This is where pumpkins would render their overlay.
+     * Called after the client camera overlay
+     * provided by the {@link DataComponents#EQUIPPABLE Equipable data component} is rendered if it's present,
+     * such as the pumpkin overlay.
```

Comment:
Currently reads 'Called after the client camera overlay provided by the Equipable data component is rendered if it's present, such as the pumpkin overlay'. It sounds weird to me since it seems like the subject is both the item and the equippable data component.

It should probably be reorganized like 'If the client camera overlay provided by the Equippable data component is present, such as for a pumpkin, this will be called after it is rendered.'